modules/python-modules/syslogng/modules/kubernetes/__init__.py

```diff
@@ -0,0 +1,95 @@
+#############################################################################
+# Copyright (c) 2022 Attila Szakacs <szakacs.attila96@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+from syslogng import Logger, LogParser
+import kubernetes
+import logging
+
+
+class KubernetesAPIEnrichment(LogParser):
+    logger = logging.getLogger(__name__)
+
+    def init(self, options):
+        self.__metadata = {}
+        self.__prefix = options["prefix"]
+
+        if options.get("in_pod", "yes") == "yes":
+            kubernetes.config.load_incluster_config()
+        else:
+            kubernetes.config.load_kube_config()
+
+        self.__client_api = kubernetes.client.CoreV1Api()
+
+        return True
+
+    def add_prefix(self, string):
+        return "{}{}".format(self.__prefix, string)
+
+    def __get_cached_pod_metadata(self, namespace_name, pod_name):
+        self.logger.debug("Trying to find cached metadata for pod {}".format(pod_name))
+        return self.__metadata[namespace_name][pod_name]
+
+    def __query_pod_metadata_from_kubernetes_api_server(self, namespace_name, pod_name):
+        self.logger.debug("Trying to query metadata for pod {} from the Kubernetes API".format(pod_name))
+        pod = self.__client_api.list_namespaced_pod(namespace_name, field_selector="metadata.name=={}".format(pod_name)).items[0]
+        container_status = pod.status.container_statuses[0]
+
+        namespace = self.__metadata.setdefault(namespace_name, {})
+        pod_metadata = namespace.setdefault(pod_name, {})
+
+        pod_metadata[self.add_prefix("pod_uuid")] = pod.metadata.uid
+
+        for label_name, label_value in pod.metadata.labels.items():
```

Comment:
This might be optional (see my other comment below).

Reply:
likewise